packages/babel-core/src/config/plugin.js

```diff
@@ -2,8 +2,8 @@
 
 export default class Plugin {
   constructor(plugin: {}, key?: string) {
-    if (plugin.name != null && typeof plugin.name !== "string") {
-      throw new Error("Plugin .name must be a string, null, or undefined");
+    if (!plugin.name || typeof plugin.name !== "string") {
+      throw new Error("Plugin .name must be a valid JS identifier of type string");
```

Comment:
@jridgewell since that constructor is taking in a plugin object with a bunch of properties, I didn't want to make the assumption that the passed in value for `name` would always be of type `string`. e.g if it received `name: true` that would still pass the first conditional